### PR TITLE
chore(tree): refactor & fix pre-merge block check

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Refactor pre-merge check into single method on `TreeExternals`. This also fixes incorrect pre-merge check on `make_canonical`.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
failing test that needs to be updated?

File: crates/blockchain-tree/src/externals.rs [hunk 1]
```diff
@@ -39,6 +42,22 @@ impl<DB, EVM> TreeExternals<DB, EVM> {
 }
 
 impl<DB: Database, EVM> TreeExternals<DB, EVM> {
+    /// Retrieves parent block total difficulty and returns `true` if the block is pre-merge.
+    pub(crate) fn is_canonical_block_pre_merge(
+        &self,
+        parent_hash: B256,
+    ) -> Result<bool, CanonicalError> {
```
Comment (gakonst) [inline crates/blockchain-tree/src/externals.rs:49]: Result<bool, _> is an anti-pattern, let's return just Result<()> and roll the `true` case in the `CanonicalError`  variant

**Comment by onbjerg** [comment]
Reopening in case it's still relevant, feel free to close again @rkrasiuk

**Comment by mattsse** [comment]
closing because we're rewriting this atm

**Pull request closed without merging**